benchmarks/dynamo/microbenchmarks/inductor_int_mm.py

```diff
@@ -0,0 +1,165 @@
+torch.backends.cuda.matmul.allow_tf32 = True
+# The flag below controls whether to allow TF32 on cuDNN. This flag defaults to True.
+torch.backends.cudnn.allow_tf32 = True
+# The flag below controls whether to allow GROUP_M to be 4 for inductor GEMMs.
+# config.matmul_allow_group_m_of_4 = True
+
+
+
+@torch._dynamo.optimize("inductor", nopython=True)
+def inductor_aten_mm(a, b):
+    return torch._int_mm(a, b)
+
+
+@torch._dynamo.optimize("inductor", nopython=True)
+def inductor_triton_mm(a, b):
+    return torch._int_mm(a, b)
+
+@torch._dynamo.optimize("inductor", nopython=True)
+def inductor_triton_bp_mm(a, b):
+    return torch._int_mm(a, b)
+
+
+
+def torch_mm(a, b):
+    return torch._int_mm(a, b)
+
+
+def triton_mm(a, b):
+    return matmul(a, b)
+
+
+def test_total_time(shapes):
+    print("| shape | torch int_mm | triton matmul | inductor aten int_mm | inductor triton int_mm | inductor triton bp int_mm |")
+    print("|--------------------------|------|------|------|------|------|")
+    for i in range(len(shapes)):
+        a_shape, b_shape = shapes[i]
+        a = torch.randint(-128, 127, a_shape, device="cuda", dtype=torch.int8)
+        b = torch.randint(-128, 127, b_shape, device="cuda", dtype=a.dtype).t().contiguous().t()
+
+        torch_ms = time_with_torch_timer(torch_mm, (a, b)).median * 1000
+        triton_ms = time_with_torch_timer(triton_mm, (a, b)).median * 1000
+
+        config.max_autotune_gemm_backends = "aten"
+        inductor_aten_mm(a, b)
+        inductor_aten_mm(a, b)
+        inductor_aten_mm(a, b)
+        ind_aten_ms = time_with_torch_timer(inductor_aten_mm, (a, b)).median * 1000
+
+        config.max_autotune_gemm_backends = "triton"
+        inductor_triton_mm(a, b)
+        inductor_triton_mm
+        inductor_triton_mm
+        ind_triton_ms = time_with_torch_timer(inductor_triton_mm, (a, b)).median * 1000
+
+        config.use_block_pointer_mm_kernel = True
+        inductor_triton_bp_mm(a,b)
+        inductor_triton_bp_mm(a,b)
+        inductor_triton_bp_mm(a,b)
+        ind_triton_bp_ms = time_with_torch_timer(inductor_triton_bp_mm, (a, b)).median * 1000
+
+        print(f"|{a_shape} x {b_shape}|{torch_ms:.4f}|{triton_ms:.4f}|{ind_aten_ms:.4f}|{ind_triton_ms:.4f}|{ind_triton_bp_ms:.4f}|")
+        torch._dynamo.reset()
+
+
+def test_GPU_time(shapes):
+    print("| shape | torch int_mm | triton matmul | inductor aten int_mm | inductor triton int_mm | inductor triton bp int_mm |")
+    print("|--------------------------|------|------|------|------|------|")
+    for i in range(len(shapes)):
+        a_shape, b_shape = shapes[i]
+
+        a = torch.randint(-128, 127, a_shape, device="cuda", dtype=torch.int8)
+        b = torch.randint(-128, 127, b_shape, device="cuda", dtype=a.dtype).t().contiguous().t()
+
+        torch_ms = triton.testing.do_bench(lambda: torch_mm(a, b))
+        triton_ms = triton.testing.do_bench(lambda: triton_mm(a, b))
```

Comment:
Seems like this one is the fastest?

Reply:
Maybe they use a lot more configs https://github.com/openai/triton/blob/main/python/triton/ops/matmul.py#L46-L67 ?